HEP-event-capture/src/cli.ts

```diff
@@ -0,0 +1,118 @@
+import { ArgumentParser } from "argparse";
+import * as _ from "lodash";
+
+import { Dhis2Metadata, Program, MetadataPayload, DataEntryForm } from "./Dhis2Metadata";
+import { Form } from "./models/Form";
+import { getResource } from "./utils";
+
+import { getUid, prettyJSON } from "./utils";
+
+function getParser(): ArgumentParser {
+    const parser = new ArgumentParser({
+        version: "0.0.1",
+        addHelp: true,
+        description: "Argparse example",
+    });
+
+    parser.addArgument(["-u", "--url"], {
+        required: true,
+        help: "DHIS2 instance URL: http[s]://username:password@server:port",
+    });
+
+    parser.addArgument(["-d", "--program-id"], {
+        required: true,
+        help: "PROGRAM_ID",
+    });
+
+    return parser;
+}
+
+async function getProgramPayload(
+    d2Metadata: Dhis2Metadata,
+    programId: string
+): Promise<MetadataPayload> {
+    const { programs } = await d2Metadata.get<{ programs: Program[] }>({
+        "programs:fields": `:all,
+            programStages[
+                :all,
+                programStageDataElements[
```

Comment:
From your comment: "The call for fetching the data might have ended up too nested and could be split into two". Nested references are ok, it simplifies the code to join them. We only avoid them when the response load is very significant for the performance (this happens when the nested elements are repeated a lot). Have you checked the sized of typical responses involved? I don't imagine this being a problem in this case.